### PR TITLE
Added functionality to clear page button in StatusModal: removes content when clicked

### PR DESCRIPTION
**Pull request by odis43**
## Notion ticket link

[Change Status Modal Clear Page Button](https://www.notion.so/uwblueprintexecs/Change-Status-Modal-Clear-Page-Button-897c498e602a475e93c96e75e95558c7?pvs=4)

## Implementation description
* When there is content in the custom input (meeting notes and worker name), the clear page button will appear
* When clicked, the aforementioned information will be deleted.
* When deleted, the clear page button will also be unloaded 

## Steps to test
1. Check if above functionality works.

## What should reviewers focus on?
* Functionality of the button

## Checklist
- [ x ] My PR name is descriptive and in imperative tense
- [ x  ] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x  ] I have run the appropriate linter(s)
- [x ] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: frontend/src/components/common/ModalComponent.tsx [hunk 1]
```diff
@@ -29,9 +29,14 @@ export type ModalProps = {
   titleColor?: string;
   mainButtonVariant?: string;
   showModalCloseButton?: boolean;
+  onClearPage?: () => void;
+  meetingNotes?: string;
+  workerName?: string;
 };
 
 const ModalComponent = ({
+  meetingNotes,
```
Comment (bowenzhu1) [inline frontend/src/components/common/ModalComponent.tsx:38]: we shouldnt pass in use case specific props like these to a generic component. 

instead you can pass in `showClearButton`, and when you call ModalComponent in StatusModal, you can pass in the logic you've defined there for `const shouldShowClearButton = meetingNotes !== "" || workerName !== "";`

File: frontend/src/components/dashboard/StatusModal.tsx [hunk 2]
```diff
@@ -36,9 +37,21 @@ const StatusModal = ({
   const [selectedOption, setSelectedOption] = useState(status);
   const [workerName, setWorkerName] = useState("");
   const [meetingNotes, setMeetingNotes] = useState("");
+
+  const hasAdditionalNotesOrWorkerName =
```
Comment (bowenzhu1) [inline frontend/src/components/dashboard/StatusModal.tsx:41]: what's this prop for ? can we remove it if its not used

File: frontend/src/components/dashboard/StatusModal.tsx [hunk 3]
```diff
@@ -10,6 +10,7 @@ import {
   Icon,
 } from "@chakra-ui/react";
 import { ChevronDown } from "react-feather";
+import { worker } from "cluster";
```
Comment (bowenzhu1) [inline frontend/src/components/dashboard/StatusModal.tsx:13]: make sure to check your imports to remove anything added that's not actually being used

**Review by bowenzhu1 — CHANGES_REQUESTED** [review]
some code changes needed, but otherwise implementation is good!

**Review by jess94yng — APPROVED** [review]

**Review by bowenzhu1 — CHANGES_REQUESTED** [review]
need to revert the files relating to case overview in this PR? don't think those changes should be here

**Review by bowenzhu1 — CHANGES_REQUESTED** [review]
this is super minor, but let's revert the naming of the case overview page back to CASEOVERVIEW_PAGE instead of CASE_OVERVIEW

**Review by bowenzhu1 — APPROVED** [review]
approving and merging

**Pull request merged**